### DISCUSSION TITLE
Mesh convergence of an EE based method

### DISCUSSION THREAD

**Discussion by shadowbourne** [wenceorg/comp4187#104] [Q&A] [answered]
When discussing mesh convergence, should we include and plot against h_t too?
As in would the x axis be h_{t}h_{x}h_{y}h_{z} or just h_{x}h_{y}h_{z}

As I see reasons for both:
* I think the overall convergence of the numerical method is determined against having h_{t}h_{x}h_{y}h_{z} as the degrees of freedom?
* However, when comparing two algorithms such as LU and EE, when one doesnt use h_t, the convergence of EE is better, when scaling h_t~h_x**2, thus making the method's "mesh" convergence far superior (when only considering mesh size).

I think overall I'm a bit lost as to the scientific standard of comparing such algorithms.

**Comment by annereinarz** [comment, marked as answer]
Generally you will plot the spatial convergence separately from the time convergence. Since the time-stepping scheme has a different convergence order than the spatial discretisation this is especially important here. To show convergence you would show both first order convergence in time (explicit Euler) and second order convergence in space (with your hopefully second order centered diffferences).

There are cases in which you would refine separately in x,y and z direction. In this case it doesn't make sense to. Just plot the overall spatial degrees of freedom against the error. As you point out the overall spatial degrees of freedom will be 1/h_{x}h_{y}h_{z}.

Make sure you either:
- use a sufficiently small time-step that it is valid for each spatial discretisation you use
- or, adjust the time step carefully

**Reply by shadowbourne** [reply]
oh ok think i understand thank you.
, however, surely any adjustment of the time step while varying spatial stepswould affect the convergence rate thus invalidating any results? (with regards to your second suggestion)

Thank you very much for your clarification

**Reply by annereinarz** [reply]
Essentially you can write the error as consisting of its time and its space component:
||u-u_h,t|| \leq ||u-u_t|| + ||u_t-u_h,t||
If you ensure that the error is dominated by one of those terms you can "neglect" the other for purposes of the convergence rate. This does need to be done carefully though. 
Coincidentally the first option can have the same issue, if the time error dominates you still will not see the expected convergence. This is unlikely to be the case for the first few entries then, but you might have issues with the last if you don't make h_t sufficiently small.

**Reply by shadowbourne** [reply]
So in this case would scaling h_t ~ h_x**2 cause this neglection you speak of..? Or would this just give a far higher convergence rate than the reality?

**Reply by annereinarz** [reply]
That's exactly the proportion you need. You won't be able to get a higher convergence rate in space by decreasing the error in the time discretisation.
